python/ml4ir/base/pipeline.py

```diff
@@ -265,9 +318,128 @@ def get_relevance_model(self, feature_layer_keys_to_fns={}) -> RelevanceModel:
         """
         raise NotImplementedError
 
+    def kfold_analysis(self, base_logs_dir, run_id, num_folds):
+        """
+        Aggregate results of the k-fold runs and perform t-test on the results.
+        """
+        raise NotImplementedError
+
+    def create_pipeline_for_kfold(self, args):
+        raise NotImplementedError
```

Comment:
Can't we have a generic definition here?
I feel this method is redundant if it always calls the respective init